src/pages/Services.tsx

```diff
@@ -2,17 +2,27 @@ import ServiceItem from '../components/ServiceItem';
 import useServices from '../hooks/useServices';
 import { IService, ServiceStatusEnum } from '../types';
 import SearchServiceButton from '../components/Form/SearchServiceButton';
+import { useState } from 'react';
+import Loading from '../components/Loading';
 
 function Services() {
+  const PAGE_SIZE = 3;
   const queryString = window.location.search;
   const searchQuery = new URLSearchParams(queryString).get('s') || undefined;
-  const services = useServices(
+  const [offset, setOffset] = useState(0);
```

Comment:
handling the offset and loadMore function here is not optimal as a practice as it add logical to manage pagination at two different places. 
To do that clean you can add to useServices a loadMore function that you can get here